packages/core/core/src/factories/useUserBillingFactory.ts

```diff
@@ -0,0 +1,96 @@
+import { Ref, computed } from '@vue/composition-api';
+import { UseUserBilling } from '../types';
+import { sharedRef } from '../utils';
+
+export interface UseUserBillingFactoryParams<ADDRESS> {
+  addAddress: (address: ADDRESS) => Promise<ADDRESS[]>;
```

Comment:
Please use single param here (like in other factories)
```
addAddress({ params })
```
this way if we add/reorder something its not a breaking change

Also you should provide access to current state here (`addressess`) and mark it as readonly.Give people everything they could need in this function from current composable.

Above applies to whole `factoryParams` object

Reply:
Ok, did it like that:
```ts
const readonlyAddresses = computed(() => addresses.value);
// ...
addresses.value = await factoryParams.addAddress({
          address,
          addresses: readonlyAddresses
        });
```